### ISSUE TITLE
Being able to add Env Vars and commands that will be enabled gloally

### ISSUE THREAD

**Issue by Toadfield** [benjamimgois/goverlay#102]
So what I mean is,that there is a extra tab in Goverlay that says something like "Custom Env Vars and Commands" and if you click on it you can click on "Add Environment Variable" or "Add Command".
So if you click on "Add Environment Variable",that you can for example set something like "MANGOHUD_DLSYM=1" and then it gets added to /.profile as "export MANGOHUD_DLSYM=1".
And if you click on "Add Command" then you can add something like "mangohud" and then that gets added to /.profile.
And you can see a list of the env vars and commands in that tab and all env vars and commands have a turn on/off switch,so if you turn it off,then it gets deleted from /.profile and vise versa.
That would be cool,cause then the devs don't have to add programs like gamscope to goverlay,but you can do it yourself.

**Comment by benjamimgois** [comment]
That's a good idea... i'll try to think about it in next release

**Comment by Toadfield** [comment]
cool

**Comment by Toadfield** [comment]
I have a better idea.
That if you add something there,that instead of that it gets added to /profile,it gets added to the "command" to all .desktop files.
So I mean that for example if you add the command "gamescope" to the global enable tab in goverlay,that then it will be placed [here](https://i.imgur.com/mCIOPmx.png) for all .desktop files you have.
I don't know how to add that feature,but it should be possible.
That's better for multiple reasons:
1.You don't have to restart your system if you add a env var or command
2.You can add things like `gamescope`,cause if you out that just in /profile,it won't work,cause only env vars work there afaik.
3.You could enable MangoHud globally for OpenGL Games,cause if you put `MANGOHUD_DLSYM=1` in /profile,it doesn't work.

**Comment by stephanlachnit** [comment]
Why don't you just use gamemode, and add these via `gamemoderun`.

**Comment by Toadfield** [comment]
> Why don't you just use gamemode, and add these via `gamemoderun`.

Isn't gamemode just for better performance?
So how would that help me there?
> Or use Lutris, which does exactly that.

If I would use Lutris,I would have to add every game,then create a application menu shortcut,then delete the old shortcuts,then search the icon and add it.
And even adding games themself takes too long cause Lutris is poorly made,you can't just drag and drop the game executable in there.
And what I want is to just install the game I want,then create a .desktop file,then launch it.

**Comment by stephanlachnit** [comment]
https://github.com/FeralInteractive/gamemode/blob/master/README.md under `Note for Hybrid GPU users`

**Comment by Toadfield** [comment]
So if I got it right,does it mean that I can use "Gamemoderun=" and then every command or env var in it?
And does that also mean that env vars that don't work globally normally would work then?

**Comment by stephanlachnit** [comment]
For example, set `/etc/environment` to
```sh
export GAMEMODERUNEXEC="env ENABLE_VKBASALT=1 mangohud --dlsym"
```
This will cause `gamemoderun vkcube` to run both with vkBasalt and MangoHud. You can do the same with gamescope.

Managing environment variables doesn't really fit into the scope of GOverlay IMHO, it's more of a config editor.

**Comment by Toadfield** [comment]
> For example, set /etc/environment to
>export GAMEMODERUNEXEC="env ENABLE_VKBASALT=1 mangohud --dlsym"
>This will cause gamemoderun vkcube to run both with vkBasalt and MangoHud. You can do the same with gamescope.

This doesn't work globally then,cause I would have to enable `gamemoderun`,or am I wrong?
>Managing environment variables doesn't really fit into the scope of GOverlay IMHO, it's more of a config editor.

I don't know a gui programming that has the ability to manage env vars.
And I think it would make sense for goverlay,cause it's still just editing files then.

**Comment by stephanlachnit** [comment]
> This doesn't work globally then,cause I would have to enable `gamemoderun`,or am I wrong?

You should make them global anyway. You don't want to enable a Vulkan overlay globally. I works right now, because only games are Vulkan. But more Vulkan applications are coming, even compositors.

> I don't know a gui programming that has the ability to manage env vars.
> And I think it would make sense for goverlay,cause it's still just editing files then.

No, that are entirely different things. On Unix everything is editing files, that's hardly an argument.

**Comment by Toadfield** [comment]
> You should make them global anyway. You don't want to enable a Vulkan overlay globally. I works right now, because only games are Vulkan. But more Vulkan applications are coming, even compositors.
But why not?
I mean I use it for most applications anyway and if I don't wanna use it for example a text editor,then I just disable it for that.
> No, that are entirely different things. On Unix everything is editing files, that's hardly an argument.

Yeah,you are right,my argument was shit,but it still would make sense if Goverlay would support that,cause,why not?

**Comment by Toadfield** [comment]
For the things I said in my one comment:

> I have a better idea.
That if you add something there,that instead of that it gets added to /profile,it gets added to the "command" to all .desktop files.
So I mean that for example if you add the command "gamescope" to the global enable tab in goverlay,that then it will be placed here for all .desktop files you have.
I don't know how to add that feature,but it should be possible.
That's better for multiple reasons:
1.You don't have to restart your system if you add a env var or command
2.You can add things like gamescope,cause if you out that just in /profile,it won't work,cause only env vars work there afaik.
3.You could enable MangoHud globally for OpenGL Games,cause if you put MANGOHUD_DLSYM=1 in /profile,it doesn't work.

Does someone have scripting knowledge and know how to achieve that?
@benjamimgois And would that be easy to implement?

**Comment by benjamimgois** [comment]
@Toadfield i agree that enabling things globally should be easier, but as @stephanlachnit said, it should be cautious. Believe me, i tried to globally enabling opengl hud, but when it was enabled it broke the desktop, because our compositor is opengl based. Adding things to the .profile should be easy to implement, but also risky. But, changing every .desktop file should be trickier and to be honest, i'm not sure it should be done.

**Comment by Toadfield** [comment]
@benjamimgois 

> i agree that enabling things globally should be easier, but as @stephanlachnit said, it should be cautious. Believe me, i tried to globally enabling opengl hud, but when it was enabled it broke the desktop, because our compositor is opengl based. Adding things to the .profile should be easy to implement, also risky.

Then I just don't add things like that to .profile and I'm fine

> But, changing every .desktop file should be trickier and to be honest, i'm not sure it should be done.

Why should it not be done?
I mean you could disable the command you added if it doesn't work for a .desktop file.

**Comment by stephanlachnit** [comment]
Because .desktop can be updated or managed by a package manager. You can add your own to .local, and if you already add it it's simple to add your env vars.

**Comment by Toadfield** [comment]
@stephanlachnit If you mean that I should make my own .desktop file,that would take so long and I still would have to enable the commands manually.

**Comment by stephanlachnit** [comment]
Simply do `alias g="<your commands herr>"` and it's literally one letter you would need to add. You can simply change the .desktop files yourself or copy them to .local, there is no benefit in wiring this up to GOverlay.

**Comment by Toadfield** [comment]
I still would have to do something.

**Comment by Toadfield** [comment]
@benjamimgois I thought about it and yeah,it would make more sense to put the env vars in .profile instead of .desktop files,cause then it wouldn't work if you just launch the executable itself.
But then putting commands in there wouldn't make sense,cause that wouldn't do anything if you do that in .profile.
